pkg/odo/cli/catalog/describe/component.go

```diff
@@ -0,0 +1,164 @@
+package describe
+
+import (
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/golang/glog"
+	"github.com/openshift/odo/pkg/catalog"
+	"github.com/openshift/odo/pkg/devfile"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/machineoutput"
+	"github.com/openshift/odo/pkg/odo/genericclioptions"
+	"github.com/openshift/odo/pkg/odo/util/experimental"
+	pkgUtil "github.com/openshift/odo/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	ktemplates "k8s.io/kubernetes/pkg/kubectl/util/templates"
+)
+
+const componentRecommendedCommandName = "component"
+
+var (
+	componentExample = ktemplates.Examples(`  # Describe a component
+    %[1]s nodejs`)
+
+	componentLongDesc = ktemplates.LongDesc(`Describe a component type.
+
+This describes the component and its' associated starter projects.
+`)
+)
+
+// DescribeComponentOptions encapsulates the options for the odo catalog describe component command
+type DescribeComponentOptions struct {
+	// name of the component to describe, from command arguments
+	componentName string
+	// if devfile components with name that matches arg[0]
+	devfileComponents []catalog.DevfileComponentType
+	// if componentName is a classic/odov1 component
+	component string
+	// generic context options common to all commands
+	*genericclioptions.Context
+}
+
+// NewDescribeComponentOptions creates a new DescribeComponentOptions instance
+func NewDescribeComponentOptions() *DescribeComponentOptions {
+	return &DescribeComponentOptions{}
+}
+
+// Complete completes DescribeComponentOptions after they've been created
+func (o *DescribeComponentOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
```

Comment:
We shouldn't name the `err` return value here. On non-OpenShift clusters, the `catalogList, err := catalog.ListComponents(o.Client)` command will return an error, and we ignore that error when experimental mode is enabled. 

However, on experimental mode, since `Complete`'s return paramter is named err, when we return, the error from ListComponents is returned, instead of nil.

Reply:
Sorry, I'm not too sure how you want to address this. Would naming the error here something else (e.g. `catalogList, catalogErr := catalog.ListComponents(o.Client)`) fix this?